cmake/FindHDF5.cmake

```diff
@@ -0,0 +1,20 @@
+pkg_check_modules(PC_HDF5 hdf5 QUIET)
```

Comment:
Should we instead use the upstream one? https://cmake.org/cmake/help/latest/module/FindHDF5.html

Reply:
We did have issues downstream unless we copied the current added `FindHDF5` at each level (conflicts with the one you linked?)

Reply:
I've never used find_dependency, only find_package. I can implement a CMake schema to find HDF5 that we've had success with for other projects. It is complicated because it matters whether HDF5 was built using CMake or Autotools and whether it was built static/shared.

Reply:
@streeve FindHDF5.cmake is part of the CMake Distributions, so if we required a high enough CMake Version, we should just have it.

Reply:
I think cmake 3.16 has the find package HDF5 which handles both HDF5 built with CMake and Autotools.

Reply:
Perfect - that's already our minimum required. So we should be able to just remove locally?

Reply:
@junghans I gave it a shot, but I'm having trouble with C/C++ difference in CMake. The CMake FindHDF5 uses `try_compile`, so I don't know a good way around globally enabling `C` or keeping a custom FindHDF5

Reply:
Is it possible to set CMAKE_C_STANDARD_REQUIRED ON and then turn it off after find package?

Reply:
Looks like it works now specifying CXX only (either required your most recent change or I did it wrong the first time)

Reply:
(continued in thread above)